### PR TITLE
Upgrade to yagna v0.10.0 and yapapi v0.9.0

### PR DESCRIPTION
**Pull request by Wiezzel**
This PR resolves following JIRA tickets

 
- [APPS-537](https://golemproject.atlassian.net/browse/APPS-537)
- [APPS-603](https://golemproject.atlassian.net/browse/APPS-603)
- [APPS-605](https://golemproject.atlassian.net/browse/APPS-605)
- [APPS-606](https://golemproject.atlassian.net/browse/APPS-606)

### Why

We want to use the latest yagna & yapapi for compatibility.

### What

* Upgraded yagna to `v0.10.0`
* Upgraded yapapi to `v0.9.0`
* Removed old hacks and cleaned up code.
* Added an option to specify a different payment driver when running the service (`erc20` was having some issues when I was debugging this)

### Assumptions 

* Have a golem box running yagna on a dedicated subnet. (Yea, it's quite difficult because we don't have enough devices, so you are excused for not testing this one. I can present a demo of it working at a standup).

### How to test

Run the app on a dedicated subnet and check if it's working:
* `YAGNA_PAYMENT_DRIVER=zksync SUBNET_TAG=<MY_SUBNET> docker-compose up`
* Go to `localhost:8000` and try to get an erigon node.

### CODE REVIEW

File: requestor/server/app.py [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
 
 
 class App(Quart):
     def __init__(self) -> None:
         super().__init__(__name__)
-        self.user_erigons: 'Mapping[str, Mapping[str, ServiceWrapper]]' = defaultdict(dict)
-        self.service_manager: 'Optional[ServiceManager]' = None
-        self.yapapi_executor_config: 'Optional[dict]' = None
+        self.user_erigons: 'MutableMapping[str, MutableMapping[str, Erigon]]' = defaultdict(dict)
+        self.golem: 'Optional[Golem]' = None
```
[the remark this answers is not in the document]
Comment (Wiezzel) [reply requestor/server/app.py:20]: As you can see below, `App` is statically initialized. I don't want to start `Golem` when importing this module.

[the remark this answers is not in the document]
Comment (johny-b) [reply requestor/server/app.py:20]: Hmm, I don't understand :/

"Creating a `Golem` instance" != "starting Golem", `await Golem.start()` should stay where it is now.

IOW, nothing important happens when initializing `Golem` object, or at least nothing should.

[the remark this answers is not in the document]
Comment (johny-b) [reply requestor/server/app.py:20]: (Don't feel obliged to answer :P)

**Review by pnowosie — APPROVED** [review]
LGTM :+1:

**Comment by Wiezzel** [comment]
@pnowosie I wanted to fix some of @johny-b 's comments... Nevermind, I'll make another PR.

File: requestor/server/app.py [hunk 2]
```diff
@@ -29,12 +29,13 @@ def __init__(self) -> None:
 
 @app.before_serving
 async def start_service_manager():
```
[the remark this answers is not in the document]
Comment (Wiezzel) [reply requestor/server/app.py:31]: Fixed in #57

File: requestor/server/app.py [hunk 3]
```diff
@@ -84,11 +85,17 @@ async def create_instance():
...
-    erigon = app.service_manager.create_service(Erigon, (init_params,), ErigonServiceWrapper)
-    erigon.name = request_data.get('name', f'erigon_{erigon.id}')
+    cluster = await app.golem.run_service(Erigon, instance_params=[{
+        "name": request_data.get('name'),
+        "init_params": init_params
+    }])
+    try:
+        erigon = cluster.instances[0]
+    except IndexError:
+        return "service initialization failed", 500
```
[the remark this answers is not in the document]
Comment (Wiezzel) [reply requestor/server/app.py:95]: Fixed in #57

**Pull request merged**